### PR TITLE
bump mockery to 2.49.2

### PR DESCRIPTION
**Pull request by paulcacheux**
### What does this PR do?

Depends on: https://github.com/DataDog/datadog-agent/pull/31022 <- the go 1.23 upgrade PR

This PR bumps mockery to latest version (2.49.2), and re-generates the mock files.
No code change, only comments.

### Motivation

This will allow us to re-enabled our "are mocks up to date?" check that was disabled in https://github.com/DataDog/datadog-agent/pull/31137, in preparation of go1.23 upgrade.

### CODE REVIEW

File: go.work [hunk 1]
```diff
@@ -1,4 +1,4 @@
-go 1.22.8
```
Comment (gjulianm) [inline go.work:1]: Do you want this go version update here?

Comment (paulcacheux) [reply go.work:1]: it's written in the PR description..
this PR depends on the go upgrade, this will be removed once I rebased after the upgrade is merged

**Review by gjulianm — APPROVED** [review]

File: go.work (hunk 1 above)
Comment (paulcacheux) [reply go.work:1]: I updated the description to make it more clear, since I expect further reviewers to ask the same question

**Review by guyarb — APPROVED** [review]

**Review by YoannGh — APPROVED** [review]

**Review by spikat — APPROVED** [review]

**Review by pgimalac — APPROVED** [review]

**Review by mbakht — APPROVED** [review]

**Review by remeh — APPROVED** [review]
LGTM

**Review by amenasria — APPROVED** [review]

**Review by kkhor-datadog — APPROVED** [review]

**Pull request merged**